section_faiss_vector_index.go

```diff
@@ -438,9 +438,16 @@ func (vo *vectorIndexOpaque) writeVectorIndexes(w *CountHashWriter) (offset uint
 	//    2. its constituent vectorID -> {docID} mapping.
 	tempBuf := vo.grabBuf(binary.MaxVarintLen64)
 	for fieldID, content := range vo.vecFieldMap {
-
-		var vecs []float32
-		var ids []int64
+		// calculate the capacity of the vecs and ids slices
+		// to avoid multiple allocations.
+		vecsLen := 0
+		idsLen := 0
+		for _, vecInfo := range content.vecs {
+			vecsLen += len(vecInfo.vec)
+			idsLen++
+		}
+		vecs := make([]float32, 0, vecsLen)
```

Comment:
shouldn't the following suffice?
```
vecs := make([]float32, 0, len(content.vecs) * content.dim)
ids := make([]int64, 0, len(content.vecs))
```

Reply:
bro that vecInfo.vec has the same len always?
like map can be like this?

ABC = [1,2,4]
XYZ = [6,7]

thts y i thought one for loop. If all vecInfo.len is same then i can remiove off this loop

Reply:
yeah all the vectors' dimensions match at this point, so its not needed